src/Glpi/Agent/Communication/AbstractRequest.php

```diff
@@ -223,6 +225,29 @@ public function handleRequest($data): bool
             }
         }
 
+        if ($auth_required === Conf::BASIC_AUTH) {
+            $authorization = $this->headers->getHeader('Authorization');
+            if (!$authorization) {
```

Comment:
```suggestion
            if (is_null($authorization)) {
```

Because `getHeader()` return `null` if related header name not exist